packages/did-auth-siop-op-authenticator/src/agent/DidAuthSiopOpAuthorizer.ts

```diff
@@ -0,0 +1,147 @@
+import { schema } from '../index'
+import { IAgentPlugin } from '@veramo/core'
+import { OpSession } from '../session/OpSession'
+import { v4 as uuidv4 } from 'uuid'
+
+import {
+  events,
+  IAuthorizeWithSiopArgs,
+  IAuthRequestDetails,
+  IRegisterSiopSessionArgs,
+  IDidAuthSiopOpAuthorizer,
+  IGetSiopAuthorizationRequestDetailsArgs,
+  IGetSiopAuthorizationRequestFromRpArgs,
+  IGetSiopSessionArgs,
+  IRegisterCustomApprovalForSiopArgs,
+  IRemoveCustomApprovalForSiopArgs,
+  IRemoveSiopSessionArgs,
+  IRequiredContext,
+  ISendSiopAuthorizationResponseArgs,
+  IVerifySiopAuthorizationRequestUriArgs,
+} from '../types/IDidAuthSiopOpAuthorizer'
+import { VerifiedAuthorizationRequest, ParsedAuthorizationRequestURI, PresentationSignCallback } from '@sphereon/did-auth-siop'
+
+export class DidAuthSiopOpAuthorizer implements IAgentPlugin {
```

Comment:
Just call it authenticator again. See previous remark about SIOP being about authentication mostly